### PR TITLE
Update Migrator for PR10 for other classes with inlined slots

### PR DESCRIPTION
**Pull request by brynnz22**
This PR updates the migrator written for PR10 (original PR for migrator: https://github.com/microbiomedata/berkeley-schema-fy24/pull/66) to look for slots that use inlined classes in the `mags_set`, `metaproteomics_analysis_set` and `metabolomics_analysis_set`. These were missed somehow when I first wrote the migrator and now comes up in the validation log as an error for missing `type` when testing the migrators. 

 It also removes a couple of lines of unnecessary and excess code.

To test this `migrator_X_to_PR10` must be run **AFTER** `migrator_X_to_PR2_and_PR24`.

### CODE REVIEW

File: nmdc_schema/migrators/migrator_from_X_to_PR10.py [hunk 1]
```diff
@@ -63,7 +63,8 @@ def upgrade(self):
 
     def add_type_to_inlined_classes(self, document: dict, slot: str, uri: str):
         r"""
-        Adds a type slot to each inlined instance of an NMDC class in the biosmpale_set, study_set, and extraction_set collections. 
+        Adds a type slot to each inlined instance of an NMDC class in the biosmpale_set, study_set, extraction_set collections, 
+        mags_set, metabolomics_analysis_set, and metaproteomics_analysis_set.
```
Comment (eecavanna) [inline nmdc_schema/migrators/migrator_from_X_to_PR10.py:67]: Now that the list is longer, I think you meant for the "collections" word to be at the very end of the comment.

File: nmdc_schema/migrators/migrator_from_X_to_PR10.py [hunk 2]
```diff
@@ -63,7 +63,8 @@ def upgrade(self):
 
     def add_type_to_inlined_classes(self, document: dict, slot: str, uri: str):
         r"""
-        Adds a type slot to each inlined instance of an NMDC class in the biosmpale_set, study_set, and extraction_set collections. 
+        Adds a type slot to each inlined instance of an NMDC class in the biosmpale_set, study_set, extraction_set collections,
```
Comment (eecavanna) [inline nmdc_schema/migrators/migrator_from_X_to_PR10.py:66]: Typo:

```diff
- biosmaple_set
+ biosample_set
```

**Review by eecavanna — APPROVED** [review]
The changes look good to me. I left comments about a couple typos.

**Comment by eecavanna** [comment]
**Side note:** 

Separately from the changes in this branch, I have a suggestion about the `add_type_slot_with_class_uri` function. The function signature currently doesn't indicate a Python data type (e.g. `int`, `str`, `dict`) for the `inlined_slots` parameter.

You can indicate that the function was designed to accept either a `dict` or `None` value for that parameter, by specifying that parameter's type as `Optional[dict]`. The `Optional` qualifier can be imported from the [`typing` module](https://docs.python.org/3.9/library/typing.html#typing.Optional), which is built into Python (i.e. it's part of the "Python standard library", like `re` and `csv`). The resulting code would look like this:

```diff
+ from typing import Optional

  # ...etc...

- def add_type_slot_with_class_uri(self, document: dict, class_uri: str, inlined_slots=None):
+ def add_type_slot_with_class_uri(self, document: dict, class_uri: str, inlined_slots: Optional[dict] = None):
```

File: nmdc_schema/migrators/migrator_from_X_to_PR10.py (hunk 1 above)
Comment (brynnz22) [reply nmdc_schema/migrators/migrator_from_X_to_PR10.py]: @eecavanna I just removed the word "collections" entirely. I think this makes sense too.

File: nmdc_schema/migrators/migrator_from_X_to_PR10.py (hunk 2 above)
Comment (brynnz22) [reply nmdc_schema/migrators/migrator_from_X_to_PR10.py]: Good catch. Fixed!

**Comment by brynnz22** [comment]
> **Side note:**
> 
> Separately from the changes in this PR branch, I have a suggestion about the `add_type_slot_with_class_uri` function. The function signature currently doesn't indicate a Python data type (e.g. `int`, `str`, `dict`) for the `inlined_slots` parameter.
> 
> You can indicate that the function was designed to accept either a `dict` or `None` value for that parameter, by specifying that parameter's type as `Optional[dict]`. The `Optional` qualifier can be imported from the [`typing` module](https://docs.python.org/3.9/library/typing.html#typing.Optional), which is built into Python (i.e. it's part of the "Python standard library", like `re` and `csv`). The resulting code would look like this:
> 
> ```diff
> + from typing import Optional
> 
>   # ...etc...
> 
> - def add_type_slot_with_class_uri(self, document: dict, class_uri: str, inlined_slots=None):
> + def add_type_slot_with_class_uri(self, document: dict, class_uri: str, inlined_slots: Optional[dict] = None):
> ```

@eecavanna thank you! I went ahead and made this change. I had no idea I could do this!

**Review by turbomam — APPROVED** [review]

**Pull request merged**